File: 4.8-bc/go/stage-2-feature-first/shared/apperr/apperr.go
// Package apperr — типізовані помилки для крос-BC помилкової семантики.
package apperr

import "fmt"

type Error struct {
	Code       string
	Message    string
	StatusCode int
}

func (e *Error) Error() string {
	return fmt.Sprintf("%s: %s", e.Code, e.Message)
}

func New(code, message string, statusCode int) *Error {
	return &Error{Code: code, Message: message, StatusCode: statusCode}
}

File: 4.8-bc/py/stage-2-feature-first/app/shared/apperr.py
"""apperr — typed errors used by every BC for cross-BC error semantics.

Each BC maps its own sentinel errors to ``AppError`` via its handler layer.
"""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class AppError(Exception):
    code: str
    message: str
    status_code: int

    def __str__(self) -> str:  # pragma: no cover
        return f"{self.code}: {self.message}"
